src/XMLHttpSource.js

```diff
@@ -69,17 +79,29 @@ XMLHttpSource.prototype = {
     paths = paths || [];
 
     var method = 'POST';
-    var queryData = [];
-    queryData.push('method=call');
-    queryData.push('callPath=' + encodeURIComponent(JSON.stringify(callPath)));
-    queryData.push('arguments=' + encodeURIComponent(JSON.stringify(args)));
-    queryData.push('pathSuffixes=' + encodeURIComponent(JSON.stringify(pathSuffix)));
-    queryData.push('paths=' + encodeURIComponent(JSON.stringify(paths)));
-
-    var queryObject = this.buildQueryObject(this._jsongUrl, method, queryData.join('&'));
-    var config = simpleExtend(queryObject, this._config);
-    config.headers["Content-Type"] = "application/x-www-form-urlencoded";
+    var config, queryData = [], queryObject;
+    if (!this._config.headers || !this._config.headers["Content-Type"] || !this._config.headers["Content-Type"].match(/application\/json/)) {
```

Comment:
This `if` condition is repeated quite a few times. Maybe there could be a `isNotJsonContent` helper for it that returns a boolean?

Reply:
I won't update this as it's master code https://github.com/Netflix/falcor-http-datasource/blob/master/src/XMLHttpSource.js#L90